### ISSUE TITLE
x/review/git-codereview: refuse to mail fixup commits?

### ISSUE THREAD

**Issue by rsc** [golang/go#20580]
I just accidentally mailed two fixup commits (e.g. https://go-review.googlesource.com/c/44857/1) instead of running git rw first. I wonder if mail should refuse to push a stack including fixup commits?

/cc @aclements

**Comment by aclements** [comment]
Seems reasonable to me, thought also not a big deal as long as they don't get submitted.

**Comment by mvdan** [comment]
While at it, you should also look for a `squash! ` prefix - that's the other form of commit that should be rebased before mailed.

**Comment by ysmolsky** [comment]
I think this was already fixed:
```
func TestDoNotMail(t *testing.T) {
	gt := newGitTest(t)
	defer gt.done()
	gt.work(t)
	trun(t, gt.client, "git", "commit", "--amend", "-m", "This is my commit.\n\nDO NOT MAIL\n")

	testMainDied(t, "mail")
	testPrintedStderr(t, "DO NOT MAIL")

	trun(t, gt.client, "git", "commit", "--amend", "-m", "fixup! This is my commit.")

	testMainDied(t, "mail")
	testPrintedStderr(t, "fixup! commit")

	trun(t, gt.client, "git", "commit", "--amend", "-m", "squash! This is my commit.")

	testMainDied(t, "mail")
	testPrintedStderr(t, "squash! commit")

	trun(t, gt.client, "git", "commit", "--amend", "-m", "This is my commit.\n\nDO NOT MAIL\n")

	// Do not mail even when the DO NOT MAIL is a parent of the thing we asked to mail.
	gt.work(t)
	testMainDied(t, "mail", "HEAD")
	testPrintedStderr(t, "DO NOT MAIL")
}
```
https://github.com/golang/review/blob/34696713d8ce9c6e709a50d8192fd973246a30e9/git-codereview/mail_test.go#L33-L58

@bradfitz, should we close it?

**Comment by aclements** [comment]
Thanks @ysmolsky. Yes, this was fixed by commit golang/review@0d52c016f0bb80623e0ce00bf4e92981bacf862e
